### PR TITLE
ref: Cache SourceMapCaches

### PR DESCRIPTION
**Pull request by loewenheim**
#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 1]
```diff
@@ -402,17 +404,23 @@ impl ArtifactFetcher {
...
+            let source_hash = Sha256::digest(&source);
+            let source_hash = BASE64.encode(&source_hash);
+            let sourcemap_hash = Sha256::digest(&sourcemap);
+            let sourcemap_hash = BASE64.encode(&sourcemap_hash);
+
+            cache_key.write_str("source:\n").unwrap();
+            cache_key.write_str(&source_hash).unwrap();
+            cache_key.write_str("\nsourcemap:\n").unwrap();
+            cache_key.write_str(&sourcemap_hash).unwrap();
+
```
Comment (kamilogorek) [inline crates/symbolicator-service/src/services/sourcemap_lookup.rs:418]: Would it be possible to write it like so? Or is there a reason you split it by section?

```rust
let cache_key_value = format!("source:\n{source_hash}\nsourcemap:\n{sourcemap_hash}");
cache_key.write_str(cache_key_value).unwrap();
```

**Review by kamilogorek — APPROVED** [review]
Left one question. #nit
Although it's Arpads specialty, so second eyes would be nice.

File: crates/symbolicator-service/src/services/caches/sourcefiles.rs [hunk 2]
```diff
@@ -40,6 +40,12 @@ impl std::ops::Deref for ByteViewString {
     }
 }
 
+impl AsRef<[u8]> for ByteViewString {
+    fn as_ref(&self) -> &[u8] {
+        self.0.get().as_ref()
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/caches/sourcefiles.rs:45]: I wonder why that doesn’t go through the existing `Deref` impl?

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 3]
```diff
@@ -402,17 +404,23 @@ impl ArtifactFetcher {
...
-        // `source` and `sourcemap`.
-        // For the time being, this all happens in a temp file that we throw away afterwards.
-        let req = FetchSourceMapCacheInternal { source, sourcemap };
-
-        let mut temp_file = self.sourcemap_caches.tempfile()?;
-        req.compute(&mut temp_file).await?;
+        let cache_key = {
+            let mut cache_key = CacheKey::scoped_builder(&self.scope);
+            let source_hash = Sha256::digest(&source);
+            let source_hash = BASE64.encode(&source_hash);
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/sourcemap_lookup.rs:410]: I would rather prefer hex formatting tbh, otherwise this looks very good!

**Review by Swatinem — APPROVED** [review]
needs a rebase now that the other PR is merged. Also using hex encoding in favor of base64 would be nice

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs (hunk 1 above)
Comment (loewenheim) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:414]: I left it as two `write`s now, hope that's ok.

**Review by Swatinem — APPROVED** [review]

**Pull request merged**